src/Core/Search/ShopFilters.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Search;
+
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Grid\Search\ShopSearchCriteriaInterface;
+
+/**
+ * ShopFilters contains a ShopConstraint to handle multishop feature, the shop constraint is mandatory and immutable.
+ * The class can't be used as is and needs to be used as a parent, you should respect the parameters orders in your
+ * implementation (if you don't you'll have to handle the specific build via a TypedBuilderInterface dedicated service).
+ */
+abstract class ShopFilters extends Filters implements ShopSearchCriteriaInterface
+{
+    /**
+     * @var ShopConstraint
+     */
+    private $shopConstraint;
+
+    /**
+     * @param ShopConstraint $shopConstraint
+     * @param array $filters
```

Comment:
shouldn't define filters array structure? I think it could show that array is associative no? (array<string, mixed>). Not sure if there are/ever wiil be filters with integers as keys :thinking: